p4-16/spec/P4-16-spec.mdk

```diff
@@ -3620,6 +3635,10 @@ parser Tcp_option_parser(packet_in b, out Tcp_option_stack vec) {
 }
 ~End P4Example
 
+Two header unions can be compared for equality (==) or inequality (!=)
+if they have the exact same type.  The unions are equal if and only if
+all their corresponding fields are equal.
```

Comment:
Only one field in a union can be "hot" so the "all" seems potentially confusing.